src/shims/x86/mod.rs

```diff
@@ -182,6 +156,53 @@ enum FloatBinOp {
     Max,
 }
 
+impl FloatBinOp {
+    /// Convert from the `imm` argument used to specify the comparison
+    /// operation in intrinsics such as `llvm.x86.sse.cmp.ss`.
+    fn cmp_from_imm(imm: i8, intrinsic: &str) -> InterpResult<'_, Self> {
+        // Bit 4 specifies whether the operation is quiet or signaling, which
+        // we do not care in Miri.
+        match imm & !0x10 {
+            // EQ_O
+            0x0 => Ok(Self::Cmp { gt: false, lt: false, eq: true, unord: false }),
```

Comment:
This sounds like it could be written with less redundacny by having a little helper `get_bit` and then using `lt: get_bit(0)` and so on?

Reply:
I reduced the table to half. It can be further reduced with some boolean logic, but I think it would make it more confusing.

Reply:
Oh, that's not nearly as nice as I had hoped... oh well, I guess I was expecting too much from Intel.^^